zip-deployboard-deck/src/snapshots/Snapshots.tsx

```diff
@@ -178,12 +191,58 @@ const SnapshotDeployModal = ({ pipeline, setPipeline, image, onClose, onSubmit }
           </Dropdown.Toggle>
           <Dropdown.Menu>
             {Array.from(SNAPSHOT_CONFIGS.values()).map((config) => (
-              <li>
-                <a onClick={() => setPipeline(config.gitBranch)}>{config.label}</a>
-              </li>
+              <Dropdown.Item>
+                <li>
+                  <a onClick={() => setPipeline(config.gitBranch)}>{config.label}</a>
+                </li>
+              </Dropdown.Item>
             ))}
           </Dropdown.Menu>
         </Dropdown>
+        {pipeline === NEXT_PREVIEW && (
+          <div
+            style={{
+              display: 'flex',
+              justifyContent: 'space-between',
+              paddingTop: '16px',
+            }}
+          >
+            <div style={{ display: 'flex', width: '70%' }}>
+              <Form>
+                <Form.Group className="mb-3" controlId="formBasicEmail">
+                  <Form.Label>Name</Form.Label>
+                  <Form.Control
+                    style={{ maxWidth: '200px' }}
+                    type="name"
+                    placeholder="Enter name"
+                    onChange={(e) => setName(e.target.value.trim())}
+                  />
+                  <Form.Text className="text-muted">You can access this at {name}.next-preview.ziphq.com</Form.Text>
+                </Form.Group>
+              </Form>
+            </div>
+            <div style={{ display: 'flex', width: '30%', flexDirection: 'column' }}>
+              <label>Select number of hours </label>
+              <Dropdown>
+                <Dropdown.Toggle>
+                  <span style={{ display: 'flex', alignItems: 'center' }}>
+                    {`Hours: ${hours}`}
+                    <ArrowDropDown />
+                  </span>
+                </Dropdown.Toggle>
+                <Dropdown.Menu>
+                  {Array.from([1, 2, 3]).map((hrs) => (
```

Comment:
maybe 0.5, 2, 4?

Reply:
I don't know if anyone would ever choose the 0.5 options, I could do 1 - 4 if that makes more sense?

Reply:
actually yeah, looking through slack it looks like people usually just take it for an hour at least, so this lgtm!